pageserver/src/tenant/timeline.rs

```diff
@@ -3015,6 +3017,37 @@ impl Timeline {
             .read_lock_held_spawn_blocking_startup_micros
             .till_now();
 
+        // before we start, because we can produce duplicate L1 layers, we must first collect all
+        // existing L1's overlapping with the LSN range, and make sure they are resident for the
+        // duration of the compaction. any duplicate Layers will end up touching the same
+        // filesystem file on eviction, and we do not want that to happen.
+        let resident_overlapping_l1s = {
+            let existing_l1s = guard
+                .layer_map()
+                .iter_historic_layers()
+                .filter(|x| !LayerMap::is_l0(x))
+                .filter(|x| {
+                    x.lsn_range.contains(&lsn_range.start)
+                        || x.lsn_range.contains(&(lsn_range.end + 1))
```

Comment:
Why the `+ 1` ? `end` is documented to be exclusive.

Reply:
I am insecure by these, unsure if I tested if it would be fine without it.